controllers/hardwareclassification_controller.go

```diff
@@ -33,24 +40,39 @@ type HardwareClassificationReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+// Reconcile reconcile function
 // +kubebuilder:rbac:groups=metal3.io,resources=hardwareclassifications,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=metal3.io,resources=hardwareclassifications/status,verbs=get;update;patch
+func (hcReconciler *HardwareClassificationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	// Initialize the logger with namespace
+	hcReconciler.Log = hcReconciler.Log.WithName(HWControllerName).WithValues("metal3-harwdwareclassification", req.NamespacedName)
 
-// Add RBAC rules to access baremetalhost resources
-// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts,verbs=get;list;update
-// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts/status,verbs=get
+	// Get HardwareClassificationController to get values for Namespace and ExpectedHardwareConfiguration
+	hardwareClassification := &hwcc.HardwareClassification{}
 
-func (r *HardwareClassificationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("hardwareclassification", req.NamespacedName)
+	if err := hcReconciler.Client.Get(ctx, req.NamespacedName, hardwareClassification); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
```

Comment:
you could use the patch helper utility from cluster API here, it is very helpful to handle the update of objects. https://github.com/kubernetes-sigs/cluster-api/blob/master/util/patch/patch.go

Reply:
you can see an example here: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/cluster_controller.go#L121 for initialization and the defer is here : https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/cluster_controller.go#L132 (see the use of the defer function)

Reply:
+1

Reply:
Added pastch helper utility in commit 
https://github.com/metal3-io/hardware-classification-controller/pull/24/commits/794187559544f6ead84444b15429f4acd9b281e6